inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java

```diff
@@ -0,0 +1,118 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Pulsar sink operator
+ */
+@Service
+public class PulsarSinkOperator extends AbstractSinkOperator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarSinkOperator.class);
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    InlongStreamEntityMapper inlongStreamEntityMapper;
+
+    @Override
+    public Boolean accept(String sinkType) {
+        return SinkType.KAFKA.equals(sinkType);
+    }
+
+    @Override
+    protected String getSinkType() {
+        return SinkType.KAFKA;
+    }
+
+    @Override
+    protected void setTargetEntity(SinkRequest request, StreamSinkEntity targetEntity) {
+        if (!this.getSinkType().equals(request.getSinkType())) {
+            throw new BusinessException(ErrorCodeEnum.SINK_TYPE_NOT_SUPPORT,
+                    ErrorCodeEnum.SINK_TYPE_NOT_SUPPORT.getMessage() + ": " + getSinkType());
+        }
+        PulsarSinkRequest sinkRequest = (PulsarSinkRequest) request;
+        try {
+            PulsarSinkDTO dto = PulsarSinkDTO.getFromRequest(sinkRequest, targetEntity.getExtParams());
+            targetEntity.setExtParams(objectMapper.writeValueAsString(dto));
+        } catch (Exception e) {
+            throw new BusinessException(ErrorCodeEnum.SINK_SAVE_FAILED,
+                    String.format("serialize extParams of Kafka SinkDTO failure: %s", e.getMessage()));
+        }
+    }
+
+    @Override
+    public StreamSink getFromEntity(StreamSinkEntity entity) {
+        PulsarSink sink = new PulsarSink();
+        if (entity == null) {
+            return sink;
+        }
+
+        PulsarSinkDTO dto = PulsarSinkDTO.getFromJson(entity.getExtParams());
+        CommonBeanUtils.copyProperties(entity, sink, true);
+        CommonBeanUtils.copyProperties(dto, sink, true);
+        List<SinkField> sinkFields = super.getSinkFields(entity.getId());
+        sink.setSinkFieldList(sinkFields);
+        return sink;
+    }
+
+    @Override
+    public Map<String, String> parse2IdParams(StreamSinkEntity streamSink, List<String> fields) {
+
+        InlongStreamEntity inlongStreamEntity = inlongStreamEntityMapper.selectByIdentifier(
+                streamSink.getInlongGroupId(), streamSink.getInlongStreamId());
+        Map<String, String> params = super.parse2IdParams(streamSink, fields);
+        PulsarSinkDTO pulsarSinkDTO;
+        try {
+            pulsarSinkDTO = objectMapper.readValue(streamSink.getExtParams(), PulsarSinkDTO.class);
+        } catch (JsonProcessingException e) {
+            LOGGER.error("parse id error", e);
+            return params;
+        }
+        params.put(PulsarParamsConfig.TOPIC, pulsarSinkDTO.getTopic());
+        params.put(PulsarParamsConfig.KEY_SEPARATOR, inlongStreamEntity.getDataSeparator());
```

Comment:
Separator and datatype are the common properties of all sinks, not only Pulsar sink's

Reply:
done